transport/http/handler.go

```diff
@@ -59,6 +59,11 @@ func (h handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if err == context.DeadlineExceeded {
+		http.Error(w, "remote timeout", http.StatusGatewayTimeout)
```

Comment:
I think we should include all the available information in this error:

    timeout for procedure %q of service %q from caller %q after %v

Then on the client side, for remote timeouts, instead of using the same `NewTimeoutError`, you'll use the 'remote' version which only accepts the full message, which is assumed to have all the necessary information.